library/src/main/java/org/apache/apex/malhar/lib/join/AbstractManagedStateInnerJoinOperator.java

```diff
@@ -72,8 +73,12 @@ public void createStores()
       stream1Store.getTimeBucketAssigner().setBucketSpan(Duration.millis(bucketSpanTime));
       stream2Store.getTimeBucketAssigner().setBucketSpan(Duration.millis(bucketSpanTime));
     }
-    stream1Store.getTimeBucketAssigner().setExpireBefore(Duration.millis(getExpiryTime()));
-    stream2Store.getTimeBucketAssigner().setExpireBefore(Duration.millis(getExpiryTime()));
+    if (stream1Store.getTimeBucketAssigner() instanceof MovingBoundaryTimeBucketAssigner) {
+      ((MovingBoundaryTimeBucketAssigner)stream1Store.getTimeBucketAssigner()).setExpireBefore(Duration.millis(getExpiryTime()));
+    }
+    if (stream2Store.getTimeBucketAssigner() instanceof MovingBoundaryTimeBucketAssigner) {
+      ((MovingBoundaryTimeBucketAssigner)stream1Store.getTimeBucketAssigner()).setExpireBefore(Duration.millis(getExpiryTime()));
+    }
 
```

Comment:
Does it make sense to allow different time bucket assigners for the stores? The stores anyways would need to be backed by a single state backend to ensure consistency, is that guaranteed?